iree/turbine/kernel/wave/constraints.py

```diff
@@ -376,6 +376,7 @@ class WorkgroupConstraint(Constraint):
     workgroup_dim: int
     apply_fn: Optional[Callable] = None
     primary: Optional[bool] = True
+    iters: Optional[IndexExpr | int] = None
```

Comment:
Could you plz explain the intention here?

My, yet uninformed, reading is that you want to override how many WG you will have along that dimension but I am unclear how this resolves with the potentially conflicting `dim/tile_size` expectation.

I see below you happen to set values that happen to agree with `dim/tile_size`, but I would love to understand the intent here (if there is something deeper that will go on).

i.e.
```
tkw.set_symbol(N_Q, seq_len)
+
  hyperparams = {
        ...
        BLOCK_N_Q: SEQ_TILE_SIZE,
```
happen to agree with:
```
tkw.WorkgroupConstraint(
  N_Q, BLOCK_N_Q, 0, iters=math.ceil(shape.max_seq_len / SEQ_TILE_SIZE)
)
```

but I assume that inserting a new `iters` arg is to support cases where they don't agree?

Reply:
Good question. Yes, in this case the sequence length is dynamic but we need to know the grid size at kernel launch. So in order to do that, we determine the number of workgroups using the maximum sequence length.